crates/sb_core/js/permissions.js

```diff
@@ -0,0 +1,279 @@
+const core = globalThis.Deno.core;
+import { pathFromURL } from "ext:deno_web/00_infra.js";
+import { Event, EventTarget } from "ext:deno_web/02_event.js";
+const primordials = globalThis.__bootstrap.primordials;
+const {
+    ArrayIsArray,
+    ArrayPrototypeIncludes,
+    ArrayPrototypeMap,
+    ArrayPrototypeSlice,
+    MapPrototypeGet,
+    MapPrototypeHas,
+    MapPrototypeSet,
+    FunctionPrototypeCall,
+    PromiseResolve,
+    PromiseReject,
+    ReflectHas,
+    SafeArrayIterator,
+    SafeMap,
+    Symbol,
+    SymbolFor,
+    TypeError,
+} = primordials;
+
+const illegalConstructorKey = Symbol("illegalConstructorKey");
+
+/**
+ * @typedef StatusCacheValue
+ * @property {PermissionState} state
+ * @property {PermissionStatus} status
+ */
+
+/** @type {ReadonlyArray<"read" | "write" | "net" | "env" | "sys" | "run" | "ffi" | "hrtime">} */
+const permissionNames = [
+    "read",
+    "write",
+    "net",
+    "env",
+    "sys",
+    "run",
+    "ffi",
+    "hrtime",
+];
+
+/**
+ * @param {Deno.PermissionDescriptor} desc
+ * @returns {Deno.PermissionState}
+ */
+function opQuery(desc) {
+    return "granted";
```

Comment:
@laktek in Deno, this line calls an op part of the `runtime`. since it was runtime specific, the quickest work-around is returning what's considered "granted". So any permission is "granted" from the edge runtime perspective. Do we want to move forward with this approach or is there something to consider?

Reply:
yeah, good for now. Let's revisit permissions handling in a future task.